src/main/java/io/openmarket/marketplace/MarketPlaceServiceHandler.java

```diff
@@ -0,0 +1,179 @@
+import io.openmarket.transaction.service.TransactionServiceHandler;
+import lombok.extern.log4j.Log4j2;
+
+import javax.annotation.Nonnull;
+import javax.inject.Inject;
+import java.util.*;
+
+import static io.openmarket.config.MerchandiseConfig.*;
+
+@Log4j2
+public class MarketPlaceServiceHandler {
+    private final ItemDao itemDao;
+    private final TransactionServiceHandler transactionServiceHandler;
+
+    //the checkout is aborted after checking transaction status for 20 second
+    private final int CHECKOUT_TIMEOUT = 20;
+    private final int CHECKOUT_QUERY_DELAY = 2000;
+    @Inject
+    public MarketPlaceServiceHandler(@Nonnull final ItemDao itemDao,
+                                     @Nonnull final TransactionServiceHandler transactionServiceHandler) {
+        this.itemDao = itemDao;
+        this.transactionServiceHandler = transactionServiceHandler;
+    }
+
+
+    public GetOrgItemsResult getListingByOrgId(GetOrgItemsRequest request) {
+        List<String> itemIds = this.itemDao.getItemIdsByOrg(request.getOrgId());
+        List<Item> selling = this.itemDao.batchLoad(itemIds);
+        List<ItemGrpc> result = new ArrayList<>();
+
+        for(Item item : selling) {
+            result.add(
+                    ItemGrpc.newBuilder()
+                    .setItemName(item.getItemName()).setItemStock(item.getStock())
+                    .setBelongTo(item.getBelongTo())
+                    .setItemPrice(item.getItemPrice())
+                    .setItemDescription(item.getItemDescription())
+                    .setItemId(item.getItemID())
+                    .setItemImageLink(item.getItemImageLink())
+                    .build()
+            );
+        }
+
+        return GetOrgItemsResult.newBuilder().addAllItems(result).build();
+    }
+
+    public CheckOutResult checkout(String uesrId, CheckOutRequest request) {
+        List<ItemGrpc> items = request.getItemsList();
+        List<Integer> itemCounts = request.getCountList();
+        double total = 0.0;
+        boolean done = false;
+        if (items.size() != itemCounts.size()  || items.isEmpty()
+        || request.getCurrencyId().isEmpty() || request.getFromOrg().isEmpty())
+            throw new IllegalArgumentException("Invalid order");
+
+
+        log.info("Checkout received from {} buying from {}",
+                uesrId, request.getFromOrg());
+
+        //puts item on hold, not finalized until transcation is processed correctly
+       CheckOutResult result = updateItemDB(items, itemCounts, null);
+
+       if (result.getUnprocessedItemCount() == items.size()) return result;
+
+       Set<ItemGrpc> unprocessed = new HashSet<>(result.getUnprocessedItemList());
+       for (int i = 0; i < items.size(); i++) {
+           ItemGrpc item = items.get(i);
+           int count = itemCounts.get(i);
+
+           if (!unprocessed.contains(item)) {
+               total += this.itemDao.load(item.getItemId()).get().getItemPrice() * count;
+           }
+       }
+
```

Comment:
How can users know about these unprocessed items?

Reply:
By writing them back to result